### ISSUE TITLE
Pagination pageNumber binding throws ExpressionChangedAfterItHasBeenCheckedError

### ISSUE THREAD

**Issue by jonest** [HealthCatalyst/Fabric.Cashmere#445]
**Describe the bug**
Pagination seems to be setting the page number to `null` before the view is rendered, throwing `ExpressionChangedAfterItHasBeenCheckedError`.

**To Reproduce**
https://stackblitz.com/edit/cashmere-demo-g5nco1?file=app/home/home.component.ts

**Expected behavior**
I expect it to not throw an error 😛 

**Screenshots**
![image](https://user-images.githubusercontent.com/3075414/44118372-15a8341a-9fd3-11e8-9645-5a5e46a943eb.png)

**Comment by jtrujill** [comment]
The issue is that the pageNumberChanged event emits a new value as soon as a new page number is set. The issue is that Angular records the input values it sets for it's child components and at the end of a digest cycle it re-runs it to see if any of those input values have changed. If they have then it indicates the possibility of there being a mismatch between the view and model. Basically the child component "hc-pagination" is changing the pageIndex of it's parent. However that same pageIndex is being used for input into hc-pagination. 

If the user used different variables for the change event and the input value to hc-pagination, this wouldn't be an issue. But we shouldn't even allow this issue to crop up so it would be better to remove the ability to use the angular notation [(pageNumber)]. We should just separate the pageNumber input and output with different names.

**Comment by jtrujill** [comment]
Another solution is to set the pageNumberChange eventEmitter to:
`@Output() readonly pageNumberChange = new EventEmitter<number | null>(true);`
which makes it async. From my understanding this will essentially create another digest cycle which eliminates the issue

**Comment by andrew-frueh** [comment]
Ah, that makes sense @jtrujill - and I believe that fix you suggested would also resolve #316

**Comment by jtrujill** [comment]
Another solution, and this might be better, is to use the changedetector method markForCheck in the set PageNumber() which would look something like: 

```
set pageNumber(value: number | null) {
        this._inputPageNumber = value;
        this.changeDetection.markForCheck();
        this.pageNumberChange.emit(value);
        this._pageNumber = this.sanitize(value);
    }
```

However looking at the source code the following lines need to be fixed.
```
this.pageNumberChange.emit(value);
this._pageNumber = this.sanitize(value);
```
The issue is that we are emitting one value but the `this.sanitize(value)` could return a different value
